backend/src/main/kotlin/ru/nsu/ctf/paketnikback/controller/FileUploadController.kt

```diff
@@ -0,0 +1,42 @@
+package ru.nsu.ctf.paketnikback.controller
+
+import io.minio.MinioClient
+import io.minio.errors.MinioException
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import org.springframework.web.multipart.MultipartFile
+
+@RestController
+@RequestMapping("/api/upload")
+class FileUploadController(
+    private val minioClient: MinioClient
+) {
+
+    @PostMapping
+    fun uploadFiles(@RequestParam("files") files: List<MultipartFile>): ResponseEntity<Map<String, String>> {
+        val uploadStatus = mutableMapOf<String, String>()
+
+        files.forEach { file ->
+            try {
+                val bucketName = "traffic-dumps" // Название бакета MinIO
+                if (!minioClient.bucketExists { it.bucket(bucketName) }) {
+                    minioClient.makeBucket { it.bucket(bucketName) }
+                }
+
+                // Сохранение файла в MinIO
+                minioClient.putObject {
+                    it.bucket(bucketName)
+                    it.`object`(file.originalFilename)
+                    it.stream(file.inputStream, file.size, -1)
+                    it.contentType(file.contentType)
+                }
+
+                uploadStatus[file.originalFilename ?: "unknown"] = "OK"
+            } catch (e: MinioException) {
+                uploadStatus[file.originalFilename ?: "unknown"] = "НЕ ОК: ${e.message}"
+            }
+        }
+
+        return ResponseEntity.ok(uploadStatus)
```

Comment:
Если какие-то загрузки не удались, вернем 200?

Предлагаю
```suggestion
    val allSuccessful = uploadStatus.values.all { it == "OK" }
    val anyFailed = uploadStatus.values.any { it != "OK" }

    val status = when {
        allSuccessful -> HttpStatus.OK
        anyFailed && !allSuccessful -> HttpStatus.PARTIAL_CONTENT
        else -> HttpStatus.BAD_REQUEST
    }

    return ResponseEntity(uploadStatus, status)
``` 

Если какие-то файлы загрузились, вернем PARTIAL_CONTENT, если все - OK, если ни один -

Reply:
Кажется, если все, то двухсотим, если не все, то можно и partial content (хотя этот ответ вроде как не об этом, но пофиг). При этом мы в ответе должны вернуть то, какие файлы не удалось загрузить. Если ни один, тут ответ зависит от того, почему не удалось загрузить. Если, например, хранилище отвалилось, тот тут пятисотить надо по идее. Не клиент же виноват, что что-то на бэке сдохло